pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -256,11 +256,11 @@ impl PdbScan {
             // quals, we'd endup scanning the whole tantivy index.
             // However, the Join quals help with scoring and snippet generation, as the documents
             // that match partially the Join quals will be scored and snippets generated. That is
-            // why it only makes sense to use the Join quals if we have used our operator.
-            // TODO(mdasht): we should make it even more restrictive and only use the Join quals if
-            // we have used our operator and also use paradedb.score and paradedb.snippet functions
-            // in the query.
-            if state.uses_our_operator {
+            // why it only makes sense to use the Join quals if we have used our operator and
+            // also used paradedb.score or paradedb.snippet functions in the query.
+            let target_list = (*(*root).parse).targetList;
+            let uses_score_or_snippet = maybe_needs_const_projections(target_list.cast());
```

Comment:
I'm not sure if this is enough though, as it covers much less than what we're checking in https://github.com/paradedb/paradedb/blob/95a8768ed803f7e42d6a2bc7a11205d9af75aa45/pg_search/src/query/mod.rs#L294-L316